ui/app/components/http-requests-bar-chart.js

```diff
@@ -89,6 +92,20 @@ export default Component.extend({
     const barChartSVG = d3.select('.http-requests-bar-chart');
     const barsContainer = d3.select('#bars-container');
```

Comment:
(+1 on this comment)

I'm not sure if this is what you're referring to, but… Ember generates a unique `id` for each component's wrapping element, which you can access from within the component as `this.element.id`. This can be used to build component-scoped selectors for things like d3.

Reply:
Ah yeah, that'll be better here instead of the `uuid`. @alisdair I don't suppose you know if you can get to that somehow when you use a component with an empty tag (so without an `element`)? I remembered where I was trying to get to it here in a little scratch/playground I was experimenting with:

https://github.com/hashicorp/consul/blob/scratch/ui-backend-search/ui-v2/app/components/event-source.js

I couldn't use the element id as I didn't have an element 🙃 , but if I could have figured out the way to access whatever creates them I would have used that.

(sorry to hijack your PR @noelledaley !)

Reply:
(Update: I was mistaken, it's `this.elementId`, not `this.element.id`)

> @alisdair I don't suppose you know if you can get to that somehow when you use a component with an empty tag (so without an element)?

Yep, you should be able to use [guidFor(this)](https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Finternals/guidFor), [which is what Ember does for components without a specified `elementId`](https://github.com/emberjs/ember.js/blob/e0be0db51d29f7739c7248ce27ec31f2961d4485/packages/%40ember/-internals/views/lib/mixins/view_support.js#L399-L401).

Reply:
Ah perfect thankyou! 👏

Reply:
good catch & thank you for all of these links and examples. very helpful! i needed to use an id to properly link the clip-path to the right element, but i updated the id to include the `elementId` so it doesn't clash if we ever use the component twice on the same page. thank you!